src/NServiceBus.RavenDB/Internal/RavenTransactionRecoveryStorageCreator.cs

```diff
@@ -0,0 +1,82 @@
+{
+    using System;
+    using System.Collections.Generic;
+    using System.Configuration;
+    using System.IO;
+    using NServiceBus.Logging;
+    using NServiceBus.Settings;
+    using Raven.Client;
+    using Raven.Client.Document;
+    using Raven.Client.Document.DTC;
+
+    class RavenTransactionRecoveryStorageCreator
+    {
+        readonly string suffixPath;
+
+        static readonly ILog logger = LogManager.GetLogger<RavenTransactionRecoveryStorageCreator>();
+
+        private RavenTransactionRecoveryStorageCreator(IDocumentStore store)
+        {
+            var docStore = store as DocumentStore;
+            if (docStore == null)
+            {
+                throw new InvalidOperationException("Cannot create transaction recovery storage for anything but a full DocumentStore instance.");
+            }
+
+            this.suffixPath = Path.Combine("NServiceBus.RavenDB", docStore.ResourceManagerId.ToString());
+        }
+
+        public static ITransactionRecoveryStorage Create(IDocumentStore store, ReadOnlySettings settings)
+        {
+            var creator = new RavenTransactionRecoveryStorageCreator(store);
+            foreach (var location in creator.TryToCreateFromConfiguredLocations(settings))
+            {
+                if (location != null)
+                    return location;
+            }
+
+            throw new InvalidOperationException($"Unable to store RavenDB transaction recovery storage information. Specify a writeable directory using the `{RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey}` appSetting or .SetTransactionRecoveryStorageBasePath(basePath) option. `%LOCALAPPDATA%`, `%APPDATA%`, or `%PROGRAMDATA%` are good candidates. The same path can be shared by multiple endpoints.");
+        }
+
+        private IEnumerable<LocalDirectoryTransactionRecoveryStorage> TryToCreateFromConfiguredLocations(ReadOnlySettings settings)
+        {
+            var settingsBasePath = settings.GetOrDefault<string>(RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey);
+            yield return TryCreate(settingsBasePath,
+                "Unable to access RavenDB transaction recovery storage specified by `.SetTransactionRecoveryStorageBasePath(string basePath)`.");
+
+            var configBasePath = ConfigurationManager.AppSettings[RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey];
+            yield return TryCreate(configBasePath,
+                $"Unable to access RavenDB transaction recovery storage specified in appSetting `{RavenDbPersistenceSettingsExtensions.TransactionRecoveryStorageBasePathKey}`.");
+        }
+
+        private LocalDirectoryTransactionRecoveryStorage TryCreate(string basePath, string errorMessage)
+        {
+            if (basePath == null)
+            {
+                return null;
+            }
+
+            var fullPath = Path.Combine(Environment.ExpandEnvironmentVariables(basePath), this.suffixPath);
```

Comment:
:-1: for this. Why not just ask the user to pass the path as is? (if we get this wrong users will loose data)